config.yaml

```diff
@@ -286,19 +286,41 @@ synchronization:
     pubsub:
         redis:
             host: "redis.demo.svc:6379"
-            password: ""
+            ## Use auth_token or redis_secret instead of password to avoid security scanning alerts
+            ## Supports: auth_token, redis_secret, password (in order of preference)
+            ## Fallback: REDIS_PASSWORD environment variable if config field is empty
+            auth_token: ""
             database: 0
             ## channel: "optimizely-sync"  # Base channel name (NOT currently parsed - uses hardcoded default)
             ## Agent publishes to channels: "optimizely-sync-{sdk_key}"
             ## For external Redis clients: Subscribe "optimizely-sync-{sdk_key}" or PSubscribe "optimizely-sync-*"
             ## Note: Channel configuration parsing is a known bug - planned for future release
+
+            ## Redis Streams configuration (when using Redis Streams for notifications)
+            ## batch_size: number of messages to batch before sending (default: 10)
+            batch_size: 5
+            ## flush_interval: maximum time to wait before sending a partial batch (default: 5s)
+            flush_interval: 2s
+            ## max_retries: maximum number of retry attempts for failed operations (default: 3)
+            max_retries: 3
+            ## retry_delay: initial delay between retry attempts (default: 100ms)
+            retry_delay: 100ms
+            ## max_retry_delay: maximum delay between retry attempts with exponential backoff (default: 5s)
+            max_retry_delay: 5s
+            ## connection_timeout: timeout for Redis connections (default: 10s)
+            connection_timeout: 10s
     ## if notification synchronization is enabled, then the active notification event-stream API
     ## will get the notifications from available replicas
     notification:
-        enable: false
-        default: "redis"
+        enable: true
+        ## Use "redis" for fire-and-forget pub/sub (existing behavior)
+        ## Use "redis-streams" for persistent message delivery with retries and acknowledgment
+        default: "redis-streams"
     ## if datafile synchronization is enabled, then for each webhook API call
     ## the datafile will be sent to all available replicas to achieve better eventual consistency
     datafile:
         enable: false
+        ## Use "redis" for fire-and-forget pub/sub (existing behavior)
+        ## Use "redis-streams" for persistent message delivery with retries and acknowledgment
         default: "redis"
+        # default: "redis-streams"  # Uncomment to enable Redis Streams
```

Comment:
datafile sync is agent-internal. Do we need to care about this redis-streams option?  
I see a value, but it may be confusing the clients.